lang/cs/Org.Apache.REEF.Bridge/DriverLauncher.cpp

```diff
@@ -0,0 +1,376 @@
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+#include "InteropUtil.h"
+using namespace std;;
+
+// Function signatuire for creating java vm
+typedef jint(JNICALL *JNI_CreateJavaVM_FN)(JavaVM **pvm, void **penv, void *args);
+
+// The environment variable that points to java directory.
+LPCTSTR JAVA_HOME = L"JAVA_HOME";
+
+// Where should we try to load jvm dll from?
+// Try server version first. Path relative to $(JAVA_HOME)
+LPCTSTR JVM_DLL1 = L"\\jre\\bin\\server\\jvm.dll";
+
+// If we could not find server jvm, Try client version.
+// Path relative to $(JAVA_HOME)
+LPCTSTR JVM_DLL2 = L"\\jre\\bin\\client\\jvm.dll";
+
+// Name of the function that creates a java VM
+const char* JNI_CreateJavaVM_Func_Name = "JNI_CreateJavaVM";
+
+// Sometimes classpath is split into 2 arguments. Compensate for it.
+const char* JavaOptionClassPath = "-classpath";
+
+// JNI signature for Java String.
+const char* JavaMainMethodSignature = "([Ljava/lang/String;)V";
+
+// class name for NativeInterop
+const char* NativeInteropClass = "org/apache/reef/javabridge/NativeInterop";
+
+const int maxPathBufSize = 16 * 1024;
+const char ClassPathSeparatorCharForWindows = ';';
+
+// we look for this to delineate java vm arguments from app arguments.
+// todo: be smarter about this. Accomodate arbitrary apps.
+const char* launcherClass = "org.apache.reef.runtime.common.REEFLauncher";
+
+// method to invoke
+const char* JavaMainMethodName = "main";
+
+//Prefix for classpath. This is how we tell JNI about the classpath.
+const char* JavaClassPath = "-Djava.class.path=";
+
+
+typedef enum {
+	ErrSuccess = 0,
+	ErrGetEnvironmentVariable = 1,
+	ErrLoadLibraryJVM = 2,
+	ErrGetProcAddress = 3,
+	ErrCreateJavaVM = 4,
+	ErrFindClassEntry = 5,
+	ErrGetStaticMethodID = 6,
+	ErrFindClassString = 7,
+	ErrNewObjectArray = 8,
+} ErrorEnum;
+
+// 
+// figure out where jvm options end, entry class method and its parameters  begin
+//
+void GetCounts(
+	int cArgs,
+	char*argv[],
+	int&  optionCount,
+	int& firstOptionOrdinal,
+	int& argCount,
+	int& firstArgOrdinal)
+{
+	bool option = true;
+	optionCount = 0;
+	firstOptionOrdinal = 2;
+	argCount = 0;
+	firstArgOrdinal = -1;
+
+	for (int i = firstOptionOrdinal; i < cArgs; i++) {
+		if (option && 0 == strcmp(argv[i], launcherClass)) {
+			option = false;
+			firstArgOrdinal = i;
+		}
+		if (option) {
+			++optionCount;
+		}
+		else {
+			++argCount;
+		}
+	}
+}
+
+//
+// Set JVM option. JNI does not support unicode
+//
+void SetOption(JavaVMOption *option, char *src)
+{
+	size_t len = strlen(src) + 1;
+	char* pszOption = new char[len];
```

Comment:
no delete, memory leak

in general i would suggest that we use std::string instead of raw chars